app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -176,4 +178,32 @@ def parse_image(image, is_public)
 
     return uid, new_result
   end
+
+  def parse_vapp_template(vapp_template)
+    uid = vapp_template.id
+
+    begin
+      # The content of the template is the OVF specification of the vApp template
+      content = @connection.get_vapp_template_ovf_descriptor(uid).body
+
+      new_result = {
+        :type        => ManageIQ::Providers::Vmware::CloudManager::OrchestrationTemplate.name,
+        :ems_ref     => uid,
+        :name        => vapp_template.name,
+        :description => vapp_template.description,
+        :orderable   => false,
+        :content     => content,
+        # By default #save_orchestration_templates_inventory does not set the EMS
+        # ID because templates are not EMS specific. We are setting the EMS
+        # explicitly here, because vapps are specific to concrete EMS.
+        :ems_id      => @ems.id
+      }
+
+      return uid, new_result
+    rescue
+      # Without content we should not create the orchestration template because
+      # these are identified by it (or its MD5)
+      $log.info("MIQ(#{self.class.name}.#{__method__}) Downloading vApp template for vApp template #{uid} failed")
```

Comment:
It's not likely to fail, but can happen in case of entwork issues. The problem I see with wrapping only the get request is that creating orchestration_template with an empty content will not produce proper results as the save inventory checks the content for potentially existing templates. If an empty content would be used, then two templates whose download has failed would be considered equal. So, even if we wrap only the call, I would still consider this an error and would not add the template.

I will see if I can get particular exception.